packages/components/src/components/stateful/Merchandize/SearchBar/SearchBar.tsx

```diff
@@ -93,9 +103,11 @@ export default function SearchBar(props: IProps): JSX.Element {
               />
             </form>
           </div>
-          {nbTopProducts === 0 && sortValue === 'category__position' && (
+          {hasProduct &&
+          nbTopProducts === 0 &&
+          sortValue === 'category__position' ? (
             <CustomNoTopProduct>{t('labelForPinProduct')}</CustomNoTopProduct>
-          )}
+          ) : null}
```

Comment:
Remove `: null` and use:
`Boolean(hasProduct &&  nbTopProducts === 0 && sortValue === 'category__position') && ...`